torch/onnx/_internal/fx/passes/type_promotion.py

```diff
@@ -1661,12 +1648,10 @@ def _run(self, *args, **kwargs) -> torch.fx.GraphModule:
         )
         assert not kwargs, "`kwargs` is not supported"
 
-        fake_tensor_mode = self._detect_fake_mode() or fake_tensor.FakeTensorMode(
-            allow_non_fake_inputs=True
-        )
         fake_args = self._fetch_fake_args()
+        fake_mode = self.fake_mode or torch._dynamo.utils.detect_fake_mode(fake_args)
```

Comment:
it is a bit tricky knowing when we have to use `torch._dynamo.utils.detect_fake_mode` or `torch.onnx._internal.fx._pass._detect_fake_mode`

Maybe we should always use one of them in the onnx exporter to make it uniform? or name our version differently to make clear the distinction, such as `_detect_fake_mode_from_fx_module`

Reply:
This check with dynamo utils is redundant. I'll remove.